File: Matrix Audio/matrix.audio.client/src/app/models/album.ts
import { Artist } from "./artist";
import { Entity } from "./entity";
import { Episode } from "./episode";

export interface Album extends Entity {

    title?: string;
    author?: string;
    authorId?: string;
    description?: string;
    imageWideSplash?: string;
    imageSquareSplash?: string;
    dateCreated?: Date;
    dateUpdated?: Date;
    status?: string; //draft, published, archived, deleted
    episodeCount?: number;
    duration?: number;
    tags?: string[];
    categories?: string[];
    likes?: number;
    commentCount?: number;
    downloadCount?: number;
    playCount?: number;
    durationInSeconds?: number;
    level?: number;
    artistId?: string;
    artistName?: string;

    artist?: Artist;
    episodes?: Episode[];

    isSelected?: boolean;
}

export interface AlbumCollection extends Entity {
    name?: string;
    description?: string;
    image?: string;
    count?: number;
    albums: Album[];
    albumIds?: { [key: string]: number };
    dateCreated?: Date;
    dateUpdated?: Date;
}

File: Matrix Audio/Matrix.Audio.Models/Album.cs
namespace Matrix.Audio.Models;

public class Album : Entity
{
    public string? Title { get; set; }
    public string? Subtitle { get; set; }
    public string? Description { get; set; }
    public string? Status { get; set; }
    public string? Artist { get; set; }
    public string? ArtistId { get; set; }
    public string? ImageWideSplashId { get; set; }
    public string? ImageSquareSplashId { get; set; }

    public List<string> Episodes { get; set; } = [];
    public List<string> Tags { get; set; } = [];
    public List<string> Categories { get; set; } = [];

    public int Level { get; set; } = 1000;
    public int Year { get; set; }
    public long DownloadCount { get; set; }
    public long PlayCount { get; set; }
    public long LikeCount { get; set; }
    public long CommentCount { get; set; }
    public int Ranking { get; set; }

    public DateTime? DateCreated { get; set; }
    public DateTime? DateUpdated { get; set; }
}
